source/Condition.h

```diff
@@ -0,0 +1,358 @@
+	// type's precision of 0 while any other types are passed
+	// through static_cast<bool>
+	explicit operator bool() const;
+
+	// Allow the Condition to be treated as its value in most contexts.
+	operator ValueType() const { return value; }
+
+
+
+private:
+	std::shared_ptr<ConditionsStore::ConditionEntry> GetEntry();
+	std::shared_ptr<ConditionsStore::ConditionEntry> EnsureEntry();
+
+	// The value of the condition, as this object sees it
+	ValueType value;
+
+	// The key of the condition; empty if this is literal
+	KeyType key;
+
+	// A shared_ptr to a ConditionsStore for this condition; empty if this is literal
+	std::shared_ptr<ConditionsStore> store;
+
+	// A weak_ptr to a ConditionEntry with the data. May be empty or expired.
+	std::weak_ptr<ConditionsStore::ConditionEntry> entry;
+};
+
+
+template <class V>
+Condition<V>::Condition(const V &value, std::shared_ptr<ConditionsStore> store, const KeyType &key)
+	: value(value), key(key), store(store)
+{
+}
+
+
+template <class V>
+Condition<V>::Condition(std::shared_ptr<ConditionsStore> store, const KeyType &key)
+	: value(), key(key), store(store)
+{
+	UpdateConditions();
+}
+
+
+// Allow construction and assignment between Condition types to
+// facilitate type conversion.
+template <class V>
+template <class V2>
+Condition<V>::Condition(const Condition<V2> &other)
+	: value(static_cast<ValueType>(other.Value())),
+	key(other.Key()),
+	store(const_cast<Condition<V2>&>(other).Store()),
+	entry(const_cast<Condition<V2>&>(other).entry)
+{
+}
+
+
+// Allow construction and assignment between Condition types to
+// facilitate type conversion.
+template <class V>
+template <class V2>
+Condition<V> &Condition<V>::operator=(const Condition<V2> &other)
+{
+	value = static_cast<ValueType>(other.Value());
+	key = other.Key();
+	store = const_cast<Condition<V2>&>(other).Store();
+	entry = const_cast<Condition<V2>&>(other).entry;
+	return *this;
+}
+
+
+// This commented-out operator may look convenient, but it causes
+// everything to break because any Condition=Condition assignment uses
+// this operator instead of the Condition=Condition implementation,
+// preventing explicit conversion of Conditions. A lot of code would
+// be simpler if this was working.
+//
+// template <class V>
+// template <class T, typename std::enable_if<std::is_arithmetic<T>::value>::type*>
+// Condition<V> &Condition<V>::operator=(const T &other)
+// {
+// 	value = static_cast<ValueType>(other);
+// 	return *this;
+// }
+
+
+
+// Update the value from a scope that contains it
+template <class V>
+const V &Condition<V>::SendValue()
+{
+	// If this was a literal, there is nothing to update.
+	if(!HasConditions())
+		return value;
+
+	std::shared_ptr<ConditionsStore::ConditionEntry> ce = EnsureEntry();
+
+	if(!ce)
+		return value;
+	else if(ce->provider)
+		*ce = value;
+	else
+		ce->value = value;
+	return value;
+}
+
+
+template <class V>
+template <class Filterer>
+const V &Condition<V>::Filter(Filterer filter)
```

Comment:
Shouldn't `Filterer` be limited to "something callable"? ...Indeed `Filterer` can be a `float`, isn't?

And even if the the above is false, probably this "thing" could be specified better by some comment.

Reply:
You don't need static assertions for everything imaginable. If someone provides an invalid type, they'll get compile-time errors.

Reply:
You're right: I was worried of ctors & implicit conversions, but I found I was wrong.

Still "filter" is a bit misleading: wouldn't "Update", "Change_to", "Apply" be a better name?

However, I leave the decision to you.

Reply:
> Still "filter" is a bit misleading: wouldn't "Update", "Change_to", "Apply" be a better name?

I think it was all Update/Updater a while back, and I did a global replace.

> However, I leave the decision to you.

I am grateful for your generosity.

This is one part of the code that will change substantially when I rework it to match the new design. Thus, the choice of type name doesn't matter much. Whatever is there now, won't be there after I rework the PR.